module/userExtension/helper.js

```diff
@@ -0,0 +1,490 @@
+				const programs = allProgramsData?.data?.data || []
+
+				if (programs.length === 0) {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
+					}
+				}
+
+				if (Array.from(allUserIds).length === 0) {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.USER_NOT_FOUND,
+					}
+				}
+
+				for (const program of programs) {
+					programIdMap[program.externalId] = program._id
+					programInfoMap[program._id.toString()] = program
+				}
+
+				// Fetch user profiles
+				const userProfileMap = {}
+				const userProfileResults = await Promise.allSettled(
+					Array.from(allUserIds).map((userId) =>
+						userService
+							.fetchProfileBasedOnUserIdOrName(tenantAndOrgInfo.tenantId, null, userId)
+							.then((result) => ({ userId, ...result }))
+					)
+				)
+
+				for (const result of userProfileResults) {
+					if (result.status === CONSTANTS.common.PROMISE_FULFILLED && result.value.success) {
+						userProfileMap[result.value.userId] = result.value.data
+					}
+				}
+
+				// Check if any user profiles were found
+				// if no user profiles were found, throw an error
+				if (Object.keys(userProfileMap).length === 0) {
+					throw {
+						status: HTTP_STATUS_CODE.bad_request.status,
+						message: CONSTANTS.apiResponses.USER_NOT_FOUND,
+					}
+				}
+
+				// Fetch user extensions
+
+				const userExtensionDocs = await userExtensionsQueries.userExtensionDocument(
+					{
+						userId: { $in: Object.values(userProfileMap).map((user) => user.id) },
+						tenantId: tenantAndOrgInfo.tenantId,
+					},
+					['userId', 'programRoleMapping']
+				)
+
+				const userExtensionMap = {}
+				for (const userExtension of userExtensionDocs) {
+					userExtensionMap[userExtension.userId] = userExtension
+				}
+
+				// Process each CSV row
+				// iterating through userRolesCSVData to process each user role
+				outerloop: for (const csvRow of userRolesCSVData) {
+					let userRole = UTILS.valueParser(csvRow)
+					userRole['_SYSTEM_ID'] = ''
+
+					try {
+						// Validate programs exist
+						if (userRole.programs && userRole.programs.length > 0) {
+							const programDocumentsArray = userRole.programs.map((p) => programIdMap[p]).filter(Boolean)
+
+							if (programDocumentsArray.length === 0) {
+								userRole['_SYSTEM_ID'] = ''
+								userRole.status = CONSTANTS.apiResponses.PROGRAM_NOT_FOUND
+								userRolesUploadedData.push(userRole)
+								continue
+							}
+						}
+
+						// Validate user exists
+						const userProfile = userProfileMap[userRole.user]
+						if (!userProfile) {
+							userRole['_SYSTEM_ID'] = ''
+							userRole.status = CONSTANTS.apiResponses.USER_NOT_FOUND
+							userRolesUploadedData.push(userRole)
+							continue outerloop
+						}
+
+						// Validate platform roles
+						const platform_role_array = userRole.platform_role?.split(',').map((r) => r.trim()) || []
```

Comment:
@borkarsaish65 instead of r can we rename to role

Reply:
done